src/components/PathwayExercise/ExerciseContent/Assessment/index.js

```diff
@@ -0,0 +1,211 @@
+                    ? answer === index + 1 && classes.correctAnswer
+                    : answer === index + 1 && classes.inCorrectAnswer
+                  : answer === index + 1 && classes.option
+              }
+              onClick={() => setAnswer(index + 1)}
+            >
+              <Typography variant="body1" sx={{ p: "16px" }}>
+                {item}
+              </Typography>
+            </Paper>
+          );
+        })}
+      </Box>
+    );
+  }
+  if (content.component === "solution") {
+    setSolution(content.value);
+  }
+  return "";
+};
+
+function Assessment({ data }) {
+  const [answer, setAnswer] = useState();
+  const [correct, setCorrect] = useState();
+  const [solution, setSolution] = useState();
+  const [submit, setSubmit] = useState();
+  const [submitDisable, setSubmitDisable] = useState();
+
+  const submitAssessment = () => {
+    setSubmit(true);
+    if (answer == solution) {
+      setCorrect(true);
+    } else {
+      setCorrect(false);
+    }
+  };
+
+  return (
+    <Container maxWidth="sm" sx={{ align: "center", m: "40px 0 58px 0" }}>
+      <Typography variant="h6" align="center">
+        Find the Output
+      </Typography>
+      {data &&
+        data.map((content) => (
+          <AssessmentContent
+            content={content}
+            answer={answer}
+            setAnswer={setAnswer}
+            setSolution={setSolution}
+            submit={submit}
+            correct={correct}
+            setSubmitDisable={setSubmitDisable}
+          />
+        ))}
+
+      <Box textAlign="center" sx={{ display: submitDisable && "none" }}>
+        <Button
+          variant="contained"
+          sx={{ width: "256px", p: "8px 16px 8px 16px" }}
+          color={answer ? "primary" : "disable"}
+          onClick={submitAssessment}
+        >
+          Submit
+        </Button>
+      </Box>
+
+      {data &&
+        data.map((content) => {
+          if (content.component === "output") {
+            return (
+              <>
+                {submit
+                  ? correct
+                    ? content.value.correct.map((content, index) => (
+                        <AssessmentContent
+                          content={content}
+                          index={index}
+                          correct={correct}
+                          submit={submit}
+                          setSubmitDisable={setSubmitDisable}
+                        />
+                      ))
+                    : content.value.incorrect.map((content, index) => (
+                        <AssessmentContent
+                          content={content}
+                          index={index}
+                          correct={correct}
+                          submit={submit}
+                          setSubmitDisable={setSubmitDisable}
+                        />
+                      ))
+                  : null}
+              </>
+            );
+          }
+        })}
```

Comment:
It's not working, It's throwing arrow on `const dataArr = content.value?[correct ? "correct" : "incorrect"]; `
So I did like this `const dataArr = content.value && correct ? content.value.correct : content.value.incorrect`

BTW how does this work?

Reply:
Sorry, I had left out the `.` before.  It should be:
```
          const dataArr = content.value?.[correct ? "correct" : "incorrect"]; 
```
The `?.` is the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#short-circuiting_evaluation).  In this case, if `content.value` is `null` or `undefined`, the expression will evaluate to `undefined`, and it won't evaluate what comes after the `.`  Otherwise, it evaluates to `content.value[correct ? "correct" : "incorrect"]`.